### PR TITLE
move benchmark build here and fix author selectors in ui.js

### PR DESCRIPTION
**Pull request by maxbeatty**
the UI JS has been living as [an example in bestiejs/benchmark.js](https://github.com/bestiejs/benchmark.js/blob/c10dd81f2e350f0c0cb2d1dc2f59a5e1bb7cf0f3/example/jsperf/ui.js) and then [built by jsperf/benchmark.js-wrapper](https://github.com/jsperf/benchmark.js-wrapper) before being [published to npm](https://www.npmjs.com/package/@mathias/benchmark.js-wrapper-jsperf) for this project to depend on. that resulted in needing to update, tag, publish, update, shrinkwrap, etc. across 3 repositories and npm to make any minor changes like `author-email` to `authorEmail`.

in this PR, I've moved the example JS here from bestiejs/benchmark.js along with most of the gulp build from jsperf/benchmark.js-wrapper so this repo can more rapidly make improvements. here's the diff between [the current benchmark test page](https://jsperf.github.io/benchmark.js-wrapper/) and new `public/_js/test.js` https://www.diffchecker.com/2vjt2isl (I only edited to pass linting so hopefully didn't break any functionality)

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      strictSSL: true
+    }
+  })
+
+	// list in reverse order since prepending
+  // use `require.resolve` so you get file path and not file contents
+  .pipe(addSrc.prepend('client/ui.js'))
+  .pipe(addSrc.prepend(require.resolve('benchmark')))
+  .pipe(addSrc.prepend(require.resolve('platform')))
+  // Use whatever version of lodash Benchmark.js is using (hopefully?)
```
Comment (maxbeatty) [inline gulpfile.js]: I _think_ this is going to resolve to lodash@3.10.x from another dependency and **not** lodash@4 [like benchmark wants](https://github.com/bestiejs/benchmark.js/blob/c10dd81f2e350f0c0cb2d1dc2f59a5e1bb7cf0f3/package.json#L21). is that ok? should we add lodash@4 as a devDep? is there a way to reference `benchmark`'s dependencies?

Comment (jdalton) [inline gulpfile.js]: Ya you can add it as a dev dep.

**Comment by maxbeatty** [comment]
@mathiasbynens @jdalton please review and merge if everything looks good

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+// mostly copy-pasta from https://github.com/jsperf/benchmark.js-wrapper/blob/c84d86d3a6a15a94a1acb2b99ec24562adfa1d27/gulpfile.js
+
+const gulp = require('gulp');
+
+const addSrc = require('gulp-add-src');
+const concat = require('gulp-concat');
+const insert = require('gulp-insert');
+const remoteSrc = require('gulp-remote-src');
+const replace = require('gulp-replace');
+// const uglify = require('gulp-uglify');
```
Comment (jdalton) [inline gulpfile.js]: Is `uglify` and its other use meant to be commented out? Is that just leftover debug?

Comment (maxbeatty) [inline gulpfile.js]: [it's currently commented out in other repo](https://github.com/jsperf/benchmark.js-wrapper/blob/c84d86d3a6a15a94a1acb2b99ec24562adfa1d27/gulpfile.js#L66) so I left it alone. it does make it easy to spot errors like in #152

Comment (jdalton) [inline gulpfile.js]: Cool. We should probably have a toggleable  production flag.

Comment (jdalton) [inline gulpfile.js]: LGTM

**Comment by mathiasbynens** [comment]
LGTM.

I think there’s still a use for a separate repo with a static HTML page to test this on, though. This can help debug the infamous “`ui` is not defined” issue.

**Pull request merged**